compiler/src/dotty/tools/dotc/ast/TreeInfo.scala

```diff
@@ -485,11 +485,27 @@ trait TypedTreeInfo extends TreeInfo[Type] { self: Trees.Instance[Type] =>
    *        Ident
    *        Select
    *        TypeApply
+   *
+   *  (2) A unary operator expression `pre.op` where `op` is one of `+`, `-`, `~`, `!`
+   *  that has a constant type `ConstantType(v)` but that is not a constant expression
+   *  (i.e. `pre` has side-effects) is translated to
+   *
+   *     { pre; v }
+   *
+   *  This avoids the situation where we have a Select node that does not have a symbol.
    */
   def constToLiteral(tree: Tree)(implicit ctx: Context): Tree = {
     val tree1 = ConstFold(tree)
     tree1.tpe.widenTermRefExpr match {
-      case ConstantType(value) if isIdempotentExpr(tree1) => Literal(value)
+      case ConstantType(value) =>
+        if (isIdempotentExpr(tree1)) Literal(value)
+        else tree1 match {
+          case Select(qual, _) if tree1.tpe.isInstanceOf[ConstantType] =>
+            // it's a unary operator; Simplify `pre.op` to `{ pre; v }` where `v` is the value of `pre.op`
```

Comment:
If we loose op before pickling, then it'll be missing from the IDE in go-to-definition/find-all-references

Reply:
Also I don't really get what's special about unary operators here ?

Reply:
They do not get a symbol. So nothing is lost when doing the transform.

Reply:
but why do they not get a symbol ?

Reply:
Because constant folding replaces the type with a ConstantType.